jormungandr-watchdog/src/service/mod.rs

```diff
@@ -0,0 +1,374 @@
+mod control;
+mod intercom;
+mod settings;
+mod state;
+mod stats;
+mod status;
+
+pub use self::{
+    control::{Control, ControlReader, Controller},
+    intercom::{
+        Intercom, IntercomMsg, IntercomReceiver, IntercomSender, IntercomStats, IntercomStatus,
+        NoIntercom,
+    },
+    settings::{NoSettings, Settings, SettingsReader, SettingsUpdater},
+    state::{NoState, State, StateHandler, StateSaver},
+    stats::Stats,
+    status::{Status, StatusReader, StatusUpdater},
+};
+use crate::watchdog::WatchdogQuery;
+use async_trait::async_trait;
+use futures_util::future::abortable;
+use std::future::Future;
+use thiserror::Error;
+use tokio::{
+    runtime::{Builder, Handle, Runtime},
+    task::JoinHandle,
+};
+use tracing_futures::Instrument as _;
+
+pub type ServiceIdentifier = &'static str;
+
+#[async_trait]
+pub trait Service: Send + Sized + 'static {
+    const SERVICE_IDENTIFIER: ServiceIdentifier;
+
+    type State: State;
+    type Settings: Settings;
+    type Intercom: IntercomMsg;
+
+    fn prepare(service_state: ServiceState<Self>) -> Self;
+
+    async fn start(self);
+}
+
+pub trait ServiceManagerTrait {
```

Comment:
Gah, please, no `Trait` traits :slightly_frowning_face:

Reply:
What do you suggest? 

* it is not meant to be implemented/used outside of `jormungandr-watchdog-derive`;
* it clashes name with the `struct ServiceManager`

Reply:
Maybe "verb" the trait, like `ManageService`.